### PR TITLE
Adding Initial Action Component

### PR DESCRIPTION
**Pull request by ccaprio97**
## What’s changing?

image

## Why?

NAC-899

## This PR has:

- [X] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [X] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

File: query-tool/src/Piipan.QueryTool.Client/Components/MatchDetail/VulnerableInvalidStatus.razor [hunk 1]
```diff
@@ -78,7 +79,7 @@
           id="invalid_no"
           name="valid_match"
           type="radio"
-          value="No"/>
+          value="No"
```
Comment (bgfoshay) [inline query-tool/src/Piipan.QueryTool.Client/Components/MatchDetail/VulnerableInvalidStatus.razor:82]: Question: Do you not need the closing tag?

Comment (ccaprio97) [reply query-tool/src/Piipan.QueryTool.Client/Components/MatchDetail/VulnerableInvalidStatus.razor:82]: I guess the browser automatically closed it but will fix

File: query-tool/src/Piipan.QueryTool.Client/Components/MatchDetail/InitialAction.razor [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+<div>
+    <h5 class="InitalActionHeader">Initial Action</h5>
+    @if (InitiatingState)
+    {
+        <form class="usa-form initalActionDropdown">
+          <label class="usa-label initalActionDropdownLabel" for="options">Initial Action Taken</label>
+          <select class="usa-select" name="options" id="options">
+            <option value>- Select -</option>
+            <option value="NoticeSent">Notice Sent</option>
+            <option value="ProvidedVerification">Client Provided Required Verification</option>
```
Comment (bgfoshay) [inline query-tool/src/Piipan.QueryTool.Client/Components/MatchDetail/InitialAction.razor:15]: Question: Is this the only option that differs between Initiating State and Matching State? Is it possible to reduce the "@if" block around just this option?

File: query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor [hunk 3]
```diff
@@ -162,68 +162,14 @@
...
-            <div class="usa-radio">
-              <input class="usa-radio__input" id="invalid_no" name="valid_match" type="radio" value="No">
-              <label class="usa-radio__label" for="invalid_no">No</label>
-            </div>
-          </fieldset>
-        </div>
-        <h4>INITIAL ACTION COMPONENT HERE</h4>
-        <h4>FINAL DISPOSITION COMPONENT HERE</h4>
-        <button class="usa-button">Save Changes</button>
+      <div diff:ignore>
```
Comment (bgfoshay) [inline query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor:172]: Nitpick: Is it possible to add a comment regarding the block being ignored?

File: query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor [hunk 4]
```diff
@@ -234,51 +180,9 @@
...
-                  <div >
-                    <dt >Participant Closing Date:</dt>
-                    <dd >N/A</dd>
-                  </div>
-                  <div >
-                    <dt >Recent Benefits Issuance Dates:</dt>
-                    <dd >01/01/2022 - 02/02/2022 03/03/2022 - 04/04/2022</dd>
-                  </div>
-                </dl>
+              <div diff:ignore >//this is ignoring a component
```
Comment (bgfoshay) [inline query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor:183]: Nitpick: similar nitpick as above. You have a comment here but could you indicate which component is being ignored?

File: query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/ResolutionFieldsTests.razor [hunk 5]
```diff
@@ -53,139 +53,18 @@
...
-                <h4>FINAL DISPOSITION COMPONENT HERE</h4>
-                <button class="usa-button">Save Changes</button>
-            </div>
-        );
+            @<div >
+  <h3 >Update information about this match for Echo Bravo</h3>
+  <p class="infoSavedText" >After the information is saved, it will appear in the
+    <b >match information</b>
+    panel.</p>
+  <div diff:ignore>
```
Comment (bgfoshay) [inline query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/ResolutionFieldsTests.razor:61]: Nitpick: comment

File: query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor [hunk 6]
```diff
@@ -365,39 +211,9 @@
...
-                    <dd >
-                      <a href="mailto:eb-test@usda.gov" >eb-test@usda.gov</a>
-                    </dd>
-                  </div>
-                  <div >
-                    <dt >Phone Number:</dt>
-                    <dd >123-456-7890</dd>
-                  </div>
-                </dl>
+              <div diff:ignore>
```
Comment (bgfoshay) [inline query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor:216]: Nitpick: comment

File: query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor [hunk 7]
```diff
@@ -290,67 +194,9 @@
...
-                  <div >
-                    <dt >Final Disposition Taken:</dt>
-                    <dd >Benefits Approved</dd>
-                  </div>
-                  <div >
-                    <dt >Benefits Start Date:</dt>
-                    <dd >02/02/2022</dd>
-                  </div>
-                </dl>
+              <div diff:ignore>
```
Comment (bgfoshay) [inline query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor:197]: Nitpick: comment

File: query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/ResolutionFieldsTests.razor [hunk 8]
```diff
@@ -53,139 +53,18 @@
...
+  </div>
+  <div diff:ignore></div>
```
Comment (bgfoshay) [inline query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/ResolutionFieldsTests.razor:63]: Nitpick: comment

**Review by bgfoshay — COMMENTED** [review]
More nitpicks than anything else. One comment around Razor logic.

File: query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor (hunk 3 above)
Comment (ccaprio97) [reply query-tool/tests/Piipan.QueryTool.Tests/Components/MatchDetail/MatchDetailTests.razor:172]: totally forgot will add

File: query-tool/src/Piipan.QueryTool.Client/Components/MatchDetail/InitialAction.razor (hunk 2 above)
Comment (ccaprio97) [reply query-tool/src/Piipan.QueryTool.Client/Components/MatchDetail/InitialAction.razor]: good catch. Fixed

**Pull request merged**